zio-cli/shared/src/main/scala/zio/cli/Options.scala

```diff
@@ -59,14 +63,19 @@ object Options {
     def aliases(names: String*): Options[A] = copy(aliases = aliases ++ names)
 
     def collect[B](message: String)(f: PartialFunction[A, B]): Options[B] =
-      copy(flagType = Map(flagType, (a: A) => f.lift(a).fold[Either[String, B]](Left(message))(Right(_))))
+      copy(optionType = Map(optionType, (a: A) => f.lift(a).fold[Either[String, B]](Left(message))(Right(_))))
 
-    def optional: Options[Option[A]] = copy(flagType = Optional(flagType))
+    def optional: Options[Option[A]] = copy(optionType = Optional(optionType))
 
-    def map[B](f: A => B): Options[B] = copy(flagType = Map(flagType, (a: A) => Right(f(a))))
+    def map[B](f: A => B): Options[B] = copy(optionType = Map(optionType, (a: A) => Right(f(a))))
 
     def mapTry[B](f: A => B): Options[B] =
-      copy(flagType = Map(flagType, (a: A) => scala.util.Try(f(a)).toEither.left.map(_.getMessage)))
+      copy(optionType = Map(optionType, (a: A) => scala.util.Try(f(a)).toEither.left.map(_.getMessage)))
+
+    def validate(args: List[String], opts: ParserOptions): IO[List[HelpDoc.Block], (List[String], A)] =
```

Comment:
I think this function here should be responsible for parsing the name / aliases, and should delegate only the parsing of the value itself to `Type#validate`. This will eliminate duplication and is also a better design decision: because the type should not know about the structure of the option as a whole, only its own structure.

So I'd consolidate the logic here and then delegate to the type validator in here.

Note that we decided on the following format for options:

 - `--foo bar` for options that take parameters
 - `-f` for options that do not take parameters (`Toggle`) and which have single-letter `name` or `aliases`. Note that we would like this to work like `wc -clfx`, i.e. if you have lots of toggles with single-letter names, then you can combine them all using only one dash.